### PR TITLE
Add swift combine extension to integrate publisher with SwiftUI

### PR DESCRIPTION
**Pull request by martingagnon**
## Description
- Make sure JustPublisher sends an OnSubscribe before sending value and completion
- Add an adapter between `org.reactiveStreams.publisher` and `Combine.Publisher`

## Motivation and Context
- Swift Combine.publisher needs a subscription before being able to receive events.
- Support SwiftUI onReceived method

## How Has This Been Tested?
In a separate project

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by marcantoinefortier** [comment]
> Make sure JustPublisher sends an OnSubscribe before sending value and completion

I feel like this should be in it's own Pull Request since it is quite an important change.

**Comment by martingagnon** [comment]
> > Make sure JustPublisher sends an OnSubscribe before sending value and completion
> 
> I feel like this should be in it's own Pull Request since it is quite an important change.

Why it is an important change ? Publishers on 0.62 were sending onSubscribe 0.63 did not, I just put back the behavior

**Comment by marcantoinefortier** [comment]
@martingagnon You're right, I thought the JustPublisher has been there for a long time. Just realised it's a fairly new class.

File: swift-extensions/combine/CombineExensions.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import Combine
+import TRIKOT_FRAMEWORK_NAME
+
+public class PublisherAdapter<T>: Combine.Publisher {
+    private let publisher: ShopCastFramework.Publisher
```
Comment (antoinelamy) [inline swift-extensions/combine/CombineExensions.swift:5]: Normal le `ShopCastFramework`? J'ai l'impression que ça devrait être `TRIKOT_FRAMEWORK_NAME`.

Comment (martingagnon) [reply swift-extensions/combine/CombineExensions.swift:5]: TRUE 🤦

File: swift-extensions/combine/CombineExensions.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    private let subscription: ShopCastFramework.Subscription
+
+    init(_ subscription: ShopCastFramework.Subscription) {
+        self.subscription = subscription
+    }
+
+    public func request(_ demand: Subscribers.Demand) {
+        if let max = demand.max {
+            subscription.request(n: Int64(max))
+        }
```
Comment (antoinelamy) [inline swift-extensions/combine/CombineExensions.swift:55]: Il faudrait gérer le else qui correspond à `Demand.unlimited`. Quelque chose comme:
```swift
else {
    subscription.request(n: Int64.max)
}
```

**Pull request merged**